andela/casablanca-cfh
# #156096154 Receive JWT on Signup

#### What does this PR do?
receive JWT on signup
#### Description of Task to be completed?
* create method that returns a token on signup
* create a route that uses the method in the step above
* add a validation middleware to validate any incoming request
* refactor code to use ES6
* write tests for the new signup route
#### How should this be manually tested?
* run `npm install` to install required packages
* create databases named `casablanca` and `casablanca-test` in your mongo shell
* use `npm run test` to verify that the tests are passing
* use `npm run start:dev` to start up the server
* send a `POST` request using postman to this endpoint: `http://127.0.0.1:3000/api/auth/signup` with values for `name`, `email`, `username`, `password` in the request body. content type should be `application/x-www-form-urlencoded`
* you should get a response object containing a token if it was successful
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?(if applicable)
#156096154
